Tribler/Main/vwxGUI/channel.py

```diff
@@ -1519,20 +1521,26 @@ def OnChange(self, event):
             self.playlistlist.Focus()
         event.Skip()
 
+    def _on_rss_feed_updated(self, subject, change_type, object_id, rss_data):
+        if rss_data['channel'].get_channel_id() != self.channel.id:
+            return
+
+        wx.CallAfter(self.RebuildRssPanel)
+
     def OnAddRss(self, event=None):
         url = self.rss_url.GetValue().strip()
         if len(url) > 0:
-            self.torrentfeed.addURL(url, self.channel.id)
-            self.RebuildRssPanel()
+            my_channel_object = self.guiutility.utility.session.lm.channel_manager.get_my_channel(self.channel.id)
+            my_channel_object.create_rss_feed(url)
 
     def OnDeleteRss(self, event):
         item = event.GetEventObject()
 
-        self.torrentfeed.deleteURL(item.url, self.channel.id)
-        self.RebuildRssPanel()
+        my_channel_object = self.guiutility.utility.session.lm.channel_manager.get_my_channel(self.channel.id)
+        my_channel_object.remove_rss_feed(item.url)
 
     def OnRefreshRss(self, event):
-        self.torrentfeed.doRefresh()
+        # TODO(lipu): fix this
```

Comment:
This will be fixed by the next PR. I have split this metadata thing into several smaller parts. It's a remainder for myself..